### PR TITLE
MFW-675: Add documentation for full type Magic PCAPs

### PR DESCRIPTION
**Pull request by arges**
This adds how to configure a bucket for full PCAPs as well as additional
instructions for collecting full type PCAPs in the existing how-to documents.

### CODE REVIEW

File: content/magic-firewall/how-to/collect-pcaps.md [hunk 1]
```diff
@@ -135,6 +181,8 @@ Once the collection is complete, you can download the PCAP by specifying the PCA
     -H 'X-Auth-Key: 00000000000' \
     --output download.pcap
 
+Only simple PCAPs can be downloaded. For a full PCAP check the bucket specified in the `destination_conf` field for a file named `pcap_<pcap_id>.pcap` once the PCAP's status is `success`.
```
Comment (nadinelyab) [inline content/magic-firewall/how-to/collect-pcaps.md]: I think we should move the second sentence to the Full Pcaps section right above. And remove the first sentence

File: content/magic-firewall/how-to/collect-pcaps.md [hunk 2]
```diff
@@ -36,11 +53,32 @@ In addition to the above fields, the JSON body can optionally filter packets by
 - (TCP/UDP) Destination port
 - IP Protocol
 
-Currently, you can only send one collect request per minute.
+Currently, you can only send one collect request per minute for simple PCAPs.You can only have one running or pending full PCAP a time.
```
Comment (nadinelyab) [inline content/magic-firewall/how-to/collect-pcaps.md]: Nit: there is a space missing after the period

File: content/magic-firewall/how-to/collect-pcaps.md [hunk 3]
```diff
@@ -7,26 +7,43 @@ meta:
...
+The PCAPs API needs both `system` and `type` to be specified in order to start a capture. A PCAP's `system` is the product or logical subsystem where packets are captured, and a PCAP's `type` is how the captured packets are built into a PCAP file.
+
+Currently, when a PCAP is requested, packets flowing at the edge through the Magic Transit system are captured, and the system is `magic-transit`. We support two types of packet captures for this system:
+- `simple`: These packets are sampled, and the sampled packets across all edge metals are collected to build a PCAP file. The packets only contain the first 160 bytes of the payload.
+- `full`: Packets are not sampled and the full packet is captured within a given point-of-presence or colo and sent to either a GCP or AWS bucket specified by the user. This may result in multiple PCAP files.
+
+{{<Aside type="note" header="Note">}}
 
-This feature is currently in an Early Access state. For access, contact your account team.
+Before starting a `full` type packet capture, you must first follow instructions for [configuring a bucket](../pcaps-bucket-setup)
```
Comment (dcpena) [inline content/magic-firewall/how-to/collect-pcaps.md]:
```suggestion
Before starting a `full` type packet capture, you must first follow instructions for [configuring a bucket](../pcaps-bucket-setup).
```

File: content/magic-firewall/how-to/collect-pcaps.md [hunk 4]
```diff
@@ -7,26 +7,43 @@ meta:
...
-- `time_limit`: The number of seconds to limit the PCAP. The number should be less than 300 seconds and cannot be set to zero.
-- `packet_limit`: The number of packets to limit the PCAP. The number should be less than 10000 and cannot be set to zero.
-- `type`: Must be `simple` as described in the above example.
-- `system`: Must be `magic-transit` as described in the above example.
+- `time_limit`: The number of seconds to limit the PCAP. The number should be less than `300` seconds and cannot be set to zero.
+- `type`: Can be `full` or `simple` as described above.
+- `system`: Must be `magic-transit` as described above.
+
+If a full `type` is specified the following additional fields are required:
+- `destination_conf`: This specifies the bucket path. For example: `gs://my-bucket`.
```
Comment (dcpena) [inline content/magic-firewall/how-to/collect-pcaps.md]:
```suggestion
- `destination_conf`: This specifies the bucket path. For example, `gs://my-bucket`.
```

File: content/magic-firewall/how-to/collect-pcaps.md [hunk 5]
```diff
@@ -7,26 +7,43 @@ meta:
...
+- `colo`: This specifies the `colo` name we want to capture packets from. For example: `ord02`.
```
Comment (dcpena) [inline content/magic-firewall/how-to/collect-pcaps.md]:
```suggestion
- `colo`: This specifies the `colo` name we want to capture packets from. For example, `ord02`.
```

File: content/magic-firewall/how-to/collect-pcaps.md [hunk 6]
```diff
@@ -7,26 +7,43 @@ meta:
...
+
+Two optional fields are available to limit the amount of packets captured:
+- `packet_limit`: The number of packets to limit the PCAP. This number should be less than `10000` and cannot be set to zero.
+- `byte_limit`: The number of bytes to limit the PCAP. This number should less than `1000000000` bytes.
```
Comment (dcpena) [inline content/magic-firewall/how-to/collect-pcaps.md]:
```suggestion
- `byte_limit`: The number of bytes to limit the PCAP. This number should be less than `1000000000` bytes.
```

File: content/magic-firewall/how-to/collect-pcaps.md (hunk 2 above)
Comment (dcpena) [inline content/magic-firewall/how-to/collect-pcaps.md]:
```suggestion
Currently, you can only send one collect request per minute for simple PCAPs, and you can only have one running or pending full PCAP at a time.
```

File: content/magic-firewall/how-to/collect-pcaps.md [hunk 7]
```diff
@@ -123,10 +161,18 @@ The response will be similar to the one received when requesting a PCAP collecti
       "messages": []
     }
 
-While the collection is ongoing, the status will be set to `pending`. Once the PCAP is ready to download, the status will change to `success` and the file is ready to download.
+While the collection is ongoing, the status will be set to `pending` or `running`. Once the PCAP is ready, the status will change to `success`.
 
 ## Download PCAP
 
+Depending on the `type` of PCAP there are multiple ways to obtain your PCAP.
```
Comment (dcpena) [inline content/magic-firewall/how-to/collect-pcaps.md:168]:
```suggestion
Depending on the `type` of PCAP there are multiple ways to obtain your PCAP.
```
```suggestion
Depending on the `type`, there are multiple ways to obtain your PCAP.
```

File: content/magic-firewall/how-to/collect-pcaps.md [hunk 8]
```diff
@@ -123,10 +161,18 @@ The response will be similar to the one received when requesting a PCAP collecti
...
+
+### Full PCAPs
+
+To obtain full PCAPs, download the files from the buckets you've specified when taking the packet capture. There may be multiple files per capture as captures may occur across multiple machines.
```
Comment (dcpena) [inline content/magic-firewall/how-to/collect-pcaps.md]:
```suggestion
To obtain full PCAPs, download the files from the buckets you specified when taking the packet capture. There may be multiple files per capture as captures may occur across multiple machines.
```

File: content/magic-firewall/how-to/collect-pcaps.md (hunk 1 above)
Comment (dcpena) [inline content/magic-firewall/how-to/collect-pcaps.md]:
```suggestion
Only simple PCAPs can be downloaded. For a full PCAP, check the bucket specified in the `destination_conf` field for a file named `pcap_<pcap_id>.pcap` once the PCAP's status is `success`.
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
+---
+title: PCAPs Bucket Setup
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:2]:
```suggestion
title: PCAPs bucket setup
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+pcx-content-type: how-to
+meta:
+    title: PCAPs Bucket Setup
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:5]:
```suggestion
    title: PCAPs bucket setup
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+---
+
+
+# Setting up a bucket
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:9]:
```suggestion
# Set up a bucket
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+AWS and GCP buckets can be configured as targets when using the PCAP API.
+
+In order to use a bucket you will need to first Enable Destinations via API. Choose either [Amazon S3](https://developers.cloudflare.com/logs/get-started/enable-destinations/aws-s3/#manage-via-api) or [Google Cloud Storage](https://developers.cloudflare.com/logs/get-started/enable-destinations/google-cloud-storage/#manage-via-api) and follow the steps for those specific services.
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:12]:
```suggestion
To use a bucket, you will need to first enable destinations via API. Choose either [Amazon S3](https://developers.cloudflare.com/logs/get-started/enable-destinations/aws-s3/#manage-via-api) or [Google Cloud Storage](https://developers.cloudflare.com/logs/get-started/enable-destinations/google-cloud-storage/#manage-via-api) and follow the steps for those specific services.
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
+At this time you cannot manage this via the Cloudflare dashboard.
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:14]: Is "this" referring to enabling destinations? If yes, I'd change this sentence a bit to "At this time, you cannot enable destinations via the Cloudflare dashboard." I'd also add it to the end of the paragraph before beginning with "To use a bucket...".

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+
+Next you will need to validate the bucket with the PCAPs API.
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:16]:
```suggestion
Next, validate the bucket with the PCAPs API.
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+
+# Validating a bucket
+
+The first step is to send an ownership challenge to a bucket to confirm you own the bucket.
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:20]:
```suggestion
First, send an ownership challenge to a bucket to confirm you own it.
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+The `bucket` field should be the URI of the bucket. For Amazon S3 it should be in the form `s3://<bucket-name>/<directory>?region=<bucket-region>` and for Google Cloud Storage in the form `gs://<bucket-name>/<directory>`.
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:21]:
```suggestion
The `bucket` field should be the URI of the bucket. For Amazon S3, it should be in the form `s3://<bucket-name>/<directory>?region=<bucket-region>` and for Google Cloud Storage in the form `gs://<bucket-name>/<directory>`.
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 17]
```diff
@@ -0,0 +1,117 @@
...
+    "destinaton_conf": "gs://bucket-test",
+    "filename": "ownership-challenge-1234.txt"
+  },
+  "success": true,
+  "errors": [],
+  "messages": []
+}
+```
+
+The response will show the filename of the ownership-challenge text file placed within the bucket. Find the file in your bucket and copy the contents of the file.
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:50]: Should `ownership-challenge` be within tick marks?

Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md:50]:
```suggestion
The response will show the filename of the ownership-challenge text file placed within the bucket. Find the file in your bucket and copy its contents.
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+
+Next validate the bucket by inserting the copied text in the `ownership-text` below:
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md]:
```suggestion
Next, validate the bucket by inserting the copied text in the `ownership-text` below:
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 19]
```diff
@@ -0,0 +1,117 @@
...
+  "messages": []
+}
+
+```
+
+If status shows `success`, the bucket is configured and ready to use.
+
+# Listing Configured Buckets
+
+To list which buckets are available for PCAPs send a request as below:
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md]:
```suggestion
To list which buckets are available for PCAPs, send a request as shown below:
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 20]
```diff
@@ -0,0 +1,117 @@
...
+    },
+  ],
+  "success": true,
+  "errors": [],
+  "messages": []
+}
+```
+
+The `status` field will show one of the following states:
+- `pending`: a challenge response has been initiated, but needs verification
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md]:
```suggestion
- `pending`: A challenge response has been initiated but needs verification.
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 21]
```diff
@@ -0,0 +1,117 @@
...
+- `failed`: the bucket has failed to be validated and cannot be used
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md]:
```suggestion
- `failed`: The bucket failed validation and cannot be used.
```

File: content/magic-firewall/how-to/pcaps-bucket-setup.md [hunk 22]
```diff
@@ -0,0 +1,117 @@
...
+- `success`: bucket is fully verified and ready to be used
```
Comment (dcpena) [inline content/magic-firewall/how-to/pcaps-bucket-setup.md]:
```suggestion
- `success`: The bucket is fully verified and ready to be used.
```

**Review by dcpena — COMMENTED** [review]
Thanks so much for this, Chris and Nadin! This looks like a ton of suggestions, but I promise they're all really small edits.

**Review by nadinelyab — APPROVED** [review]

**Review by dcpena — APPROVED** [review]

**Pull request merged**